File: AtCoder/abc/abc238/c.cpp
#include <bits/stdc++.h>
#include <atcoder/all>
#define rep(i, n) for(int i = 0; i < (n); ++i)
#define rep1(i, n) for(int i = 1; i < (n); ++i)
using namespace std;
using namespace atcoder;
using ll = long long;
using mint = modint998244353;


int main(){
    cin.tie(nullptr);
    ios::sync_with_stdio(false);

    ll n;
    cin >> n;

    ll tmp = n;
    int d = 0;
    while(tmp){
        ++d;
        tmp /= 10;
    }

    mint res = 0;
    mint now = 9;
    rep1(i, d){
        res += (1 + now) * now / 2;
        now *= 10;
    }

    ll sub = 1;
    rep1(i, d) sub *= 10;
    mint remain = n - sub;
    
    res += (1 + remain+1) * (remain+1) / 2;

    cout << res.val() << endl;

    return 0;
}

File: AtCoder/abc/abc238/c.py
n = int(input())
mod = 998244353
d = len(str(n))

res = 0
now = 9
for i in range(1, d):
    res += (1 + now) * now // 2
    res %= mod
    now *= 10

remain = n - 10**(d-1)

res += (remain + 2) * (remain +  1) // 2

print(res % mod)
